### ISSUE TITLE
Translator rights don't seem to work

### ISSUE THREAD

**Issue by flodolo** [mozilla/pontoon#3262]
CC @canova 

I've added a new project for Turkisk (Firefox Profiler), and added [Nazim](https://pontoon.mozilla.org/contributors/canova/) as a translator. For some reason, he doesn't get translator rights.

I've checked the database, and everything looks correct (he's added to the group).

I even tried adding him to the full Translators group, and still no luck.

I think we have plenty of locales with permissions per-project, and we didn't get bug reports. Is this a caching problem?

**Comment by flodolo** [comment]
One more piece of info: Nazim shows up as [contributor in the main contributors page](https://pontoon.mozilla.org/tr/contributors/?period=1), but as translator in the [project's contributors page](https://pontoon.mozilla.org/tr/firefox-profiler/contributors/), so that seems correct.

In the UI he still can only see the suggest button, and in the cog menu there is no `Make suggestions` option.

**Comment by canova** [comment]
Yes, this is a screenshot to give you some context:
<img width="1255" alt="Screenshot 2024-06-26 at 7 50 01 PM" src="https://github.com/mozilla/pontoon/assets/466239/37508aa5-d36c-47dc-93fe-dd929b30078f">

There is no "Make suggestions" option and I can only see a "suggest" button.

**Comment by eemeli** [comment]
@canova Could you try logging out of Pontoon and back in again?

**Comment by flodolo** [comment]
Should have mentioned that we tried both log in/log out, and a different browser, with no changes.

**Comment by flodolo** [comment]
Likely another case just reported on Matrix: this user was set as [global translator](https://pontoon.mozilla.org/contributors/noernova/) for `shn` about 4 hours ago. The permissions seems to work on some projects (Terminology), but not on Common Voice.

**Comment by NoerNova** [comment]
Yes, I faced this same case on ```shn```, only in Common Voice project.

Another issue is on Teams main page when click on Unreviewed I can see the Approve button, but when click to Approve it takes forever to load. 
<img width="1504" alt="Screenshot 2567-06-27 at 01 34 36" src="https://github.com/mozilla/pontoon/assets/9565672/96e32497-de23-466e-b12c-a55fce4e898e">
<img width="1504" alt="Screenshot 2567-06-27 at 01 33 12" src="https://github.com/mozilla/pontoon/assets/9565672/447f4bc7-2b0a-431c-a794-054a759f0ac8">

**Comment by canova** [comment]
FWIW, I still can't see/use my permissions after 15 hours or so. It sounds less likely to be a server-side caching problem.

**Comment by flodolo** [comment]
Just to confirm, I just used a test account and set it as translator for profile in Turkish, I only see suggest.

@mathjazz is out until next Tuesday, but this is a P1 as far as I'm concerned.

**Comment by eemeli** [comment]
This seems to be something funky with the project+locale permissions. I’ve managed to recreate the issue locally by taking a fresh DB dump from prod. If a locale has a project-specific config, it seems like an account won't get any translator permissions in that project, even if they're set as a translator for the whole locale.

Removing the project-specific config makes translators for the locale have the appropriate rights in that project.

**Comment by eemeli** [comment]
I've identified at least the immediate cause of the problem: The `guardian_groupobjectpermission` table in the DB does not appear to get sync'd properly, and it doesn't appear to reflect the info in `auth_user_groups`.

My current suspicion is that when a locale is added to a project, the guardian table should have a row added for that relationship, but that this doesn't exist for recent-ish locale additions. If then that project+locale combo has custom translators set for it, they won't show up for anyone at all.

I've not tracked exactly since when this has been broken, but I think it's sometime before 2024-04-04, which is when the first en-CA translations of Firefox Profiler were written to git.

To confirm my suspicions on this, it should be sufficient to see if the problem replicates with `en-CA/firefox-profiler/`, but not with `en-GB/firefox-profiler/`, which was added much earlier, and which does have an appropriate guardian table entry.

It's late; I'll continue from here tomorrow.

**Comment by NoerNova** [comment]
I'm just checking by now, everything works fine for me now, Thank you.

**Comment by flodolo** [comment]
> I'm just checking by now, everything works fine for me now, Thank you.

At this point, that's because I removed the custom project setting for CV, as they were identical to the main permissions.

**Comment by eemeli** [comment]
I was able to confirm my earlier suspicions about the immediate cause, and further that the permissions are currently broken on pontoon.mozilla.org for the following project/locale pairs:

- common-voice/nqo
- firefox-profiler/fur
- firefox-multi-account-containers/fur
- firefox-profiler/en-CA
- firefox-bridge/it
- firefox-bridge/de
- firefox-bridge/fr

When `nqo` was added, its permissions were correctly initialized for the terminology, tutorial & pontoon-intro projects, but not common-voice.

Based on the dates of when we deployed what to production and the latest date I could track down for the preceding valid addition, here's an initial [regression range](https://github.com/mozilla/pontoon/compare/2a028724...80c83c09).

**Comment by eemeli** [comment]
Ok, I think I have it: The culprit is almost certainly [this change](https://github.com/mozilla/pontoon/commit/ee29b881b24d6382ec6fbbad84d3779cd4c5b105#diff-59a3c6ed5fd2ee68bb44d1de5c2309a54c3073badafc09eea454b53a09c17689R188-R190) from #3160, which did not take into account the `create_project_locale_permissions_groups` and `assign_project_locale_group_permissions` [signal receivers](https://github.com/mozilla/pontoon/blob/main/pontoon/base/signals.py). With `bulk_create`, the `pre_save` and `post_save` signals are not sent, unlike with the preceding `get_or_create`.

**Comment by canova** [comment]
Thanks for the fix! I can also verify that it's now fixed on my end.
